### PR TITLE
lib/path: clean up path functions

### PR DESCRIPTION
**Pull request by stoeckmann**
- Avoid an out of boundary access on illegal /sys files
- Better error handling
- Remove ul_prefix_fopen
- Fix typos

### CODE REVIEW

File: sys-utils/lscpu-virt.c [hunk 1]
```diff
@@ -551,7 +551,7 @@ struct lscpu_virt *lscpu_read_virtualization(struct lscpu_cxt *cxt)
 		if (virt->vendor == VIRT_VENDOR_XEN) {
 			uint32_t features;
 
-			fd = ul_prefix_fopen(cxt->prefix, "r", _PATH_SYS_HYP_FEATURES);
+			fd = ul_path_fopen(cxt->rootfs, "r", _PATH_SYS_HYP_FEATURES);
```
[the remark this answers is not in the document]
Comment (t-8ch) [reply sys-utils/lscpu-virt.c:554]: And how has this ever worked?
The `mode` and `path` arguments are passed in the wrong order...

**Review by t-8ch — CHANGES_REQUESTED** [review]

File: lib/path.c [hunk 2]
```diff
@@ -1053,12 +1034,14 @@ static int ul_path_cpuparse(struct path_cxt *pc, cpu_set_t **set, int maxcpus, i
 
 	if (islist) {
 		if (cpulist_parse(buf, *set, setsize, 0)) {
-			rc = -EINVAL;
+			errno = EINVAL;
```
[the remark this answers is not in the document]
Comment (stoeckmann) [reply lib/path.c:1037]: The `errno` variable is set by `malloc` which is called within `cpuset_alloc` so I would keep it the way it is. The system library will know best why the allocation failed.

File: sys-utils/lscpu-virt.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (stoeckmann) [reply sys-utils/lscpu-virt.c:554]: Oh, nice catch. :D

File: lib/path.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (t-8ch) [reply lib/path.c:1037]: For `malloc` indeed. `fgets` on the other hand does not.
See the discussion in #1821.

[the remark this answers is not in the document]
Comment (t-8ch) [reply lib/path.c:1037]: If errors are communicated in two different ways, at least those should give consistent values.
So on `malloc` failure the function should return `-errno`.

Or we rework it to only use one error mechanism, there aren't many users of this function.

[the remark this answers is not in the document]
Comment (karelzak) [reply lib/path.c:1037]: Please, keep it return <0 as error, it's almost everywhere as standard in the code.

File: sys-utils/lscpu-virt.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (stoeckmann) [reply sys-utils/lscpu-virt.c]: We cannot replace all fscanf calls due to a while-loop, but I have adjusted the other ones.

File: lib/path.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (stoeckmann) [reply lib/path.c:1042]: I have added an explicit errno assignment in case of fgets failure

**Review by t-8ch — APPROVED** [review]
Thanks!

**Pull request merged**